### PR TITLE
chore: Drop support for legacy Python 2

### PR DESCRIPTION
**Pull request by hugovk**
Fixes https://github.com/4Catalyzer/flask-resty/issues/255.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -30,7 +28,3 @@ script:
 
 after_success:
   - codecov
-
-branches:
```
Comment (sloria) [inline .travis.yml]: Why was this removed?

Comment (hugovk) [reply .travis.yml]: It was removed so I could test my feature branch on Travis CI and be sure it was already passing, I didn't want to submit a broken PR to you! 

I can put it restore it if you prefer, although I would recommend leaving it in so others (and you!) can also test their feature branches too.

Comment (taion) [reply .travis.yml]: I don't mind WIP PRs, actually. I added this specifically so I wouldn't see two Travis builds on my own PR branches.

File: tox.ini [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tox]
 envlist = 
-  py{27,35,37}-{base,full}-marshmallow2
+  py{35,37}-{base,full}-marshmallow2
```
Comment (taion) [inline tox.ini]:
```suggestion
  py{35,37}-{base,full}-marshmallow{2,3}
```

File: tox.ini [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
   py{35,37}-{base,full}-marshmallow3
```
Comment (taion) [inline tox.ini]:
```suggestion
```

File: tox.ini (hunk 2 above)
Comment (taion) [reply tox.ini]: err, is this valid? @sloria

Comment (sloria) [reply tox.ini]: Yep. `tox -l` shows the correct output:

```
py35-base-marshmallow2
py35-base-marshmallow3
py35-full-marshmallow2
py35-full-marshmallow3
py37-base-marshmallow2
py37-base-marshmallow3
py37-full-marshmallow2
py37-full-marshmallow3
```

**Comment by hugovk** [comment]
Updated!

**Comment by taion** [comment]
Thanks so much!

**Comment by hugovk** [comment]
You're welcome! Onwards with Python 3!
